golos.publication/golos.publication.cpp

```diff
@@ -359,21 +359,34 @@ int64_t publication::pay_curators(name author, uint64_t msgid, int64_t max_rewar
     return unclaimed_rewards;
 }
 
-void publication::use_charge(tables::limit_table& lims, structures::limitparams::act_t act, name issuer,
-                            name account, int64_t eff_vesting, symbol_code token_code, bool vestpayment, elaf_t weight) {
+int16_t publication::use_charge(tables::limit_table& lims, structures::limitparams::act_t act, name issuer,
+                            name account, int64_t eff_vesting, symbol_code token_code, bool vestpayment, int16_t weight) {
     auto lim_itr = lims.find(act);
     eosio_assert(lim_itr != lims.end(), "publication::use_charge: limit parameters not set");
     eosio_assert(eff_vesting >= lim_itr->min_vesting, "insufficient effective vesting amount");
+
+    if (act == structures::limitparams::VOTE) {
+        print("@@ ", account.value);
+        auto current_power = charge::get_current_value(config::charge_name, account, token_code, lim_itr->charge_id);
+        int16_t charge = config::_100percent - current_power;
+        print("\n weight: ", weight);
+
+        weight = ( ((abs(weight) * charge) / config::_100percent)  + 200 - 1) / 200;
+        print("\n charge: ", weight);
+    }
+
     if(lim_itr->price >= 0)
         INLINE_ACTION_SENDER(charge, use) (config::charge_name,
             {issuer, config::invoice_name}, {
                 account,
                 token_code,
                 lim_itr->charge_id,
-                int_cast(elai_t(lim_itr->price) * weight),
+                (lim_itr->price * weight) / config::_100percent,
                 lim_itr->cutoff,
-                vestpayment ? int_cast(elai_t(lim_itr->vesting_price) * weight) : 0
+                vestpayment ? (lim_itr->vesting_price * weight) / config::_100percent : 0
             });
+
+    return weight;
```

Comment:
divide this value to `k`, it should be in `% / k` units.
actually this required only if `act == structures::limitparams::VOTE`, so propose to set return value inside that `if`